### PR TITLE
[CMake] Simplify two binary variables into one tri-state variable

### PR DESCRIPTION
**Pull request by davezarzycki**
Also remove some ancient logic to detect and ignore requests to use LLD. If people want to explicitly use LLD, they probably have a reason and we shouldn't second guess them.

### CODE REVIEW

**Comment by davezarzycki** [comment]
@swift-ci please test

File: stdlib/cmake/modules/AddSwiftStdlib.cmake [hunk 1]
```diff
@@ -415,13 +415,11 @@ function(_add_target_variant_link_flags)
   endif()
 
   if(NOT SWIFT_COMPILER_IS_MSVC_LIKE)
-    # FIXME: On Apple platforms, find_program needs to look for "ld64.lld"
-    find_program(LDLLD_PATH "ld.lld")
-    if((SWIFT_ENABLE_LLD_LINKER AND LDLLD_PATH AND NOT APPLE) OR
+    if((SWIFT_USE_LINKER STREQUAL "lld") OR
```
Comment (compnerd) [inline stdlib/cmake/modules/AddSwiftStdlib.cmake:418]: I think that we should simplify this like the previous case in `cmake/modules/AddSwift.cmake`

**Review by compnerd — APPROVED** [review]
Making the Swift build system more similar to the LLVM options is definitely a good idea.

File: stdlib/cmake/modules/AddSwiftStdlib.cmake [hunk 2]
```diff
@@ -415,13 +415,11 @@ function(_add_target_variant_link_flags)
...
        ("${LFLAGS_SDK}" STREQUAL "WINDOWS" AND
         NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "WINDOWS"))
       list(APPEND result "-fuse-ld=lld")
-    elseif(SWIFT_ENABLE_GOLD_LINKER AND
+    elseif(SWIFT_USE_LINKER STREQUAL "gold" AND
```
Comment (drodriguez) [inline stdlib/cmake/modules/AddSwiftStdlib.cmake:422]: I think the manipulations around these lines (L418-L422) are the same to the ones in `AddSwiftUnittests.cmake`, but the modifications are different: some checks are kept here, while they are drop in the other file. Is the divergence intentional?

**Review by drodriguez — APPROVED** [review]
The parts that I know (the Android parts, mostly), looks fine to me. Should be functionally equivalent to the existing code. There's only that one point that should be double checked. Otherwise looks good.

**Comment by davezarzycki** [comment]
Hi @compnerd and @drodriguez – I've simplified AddSwift.cmake more. I think we should make `SWIFT_USE_LINKER` be as much like `LLVM_USE_LINKER` as possible. In other words, if it's set, trust it.

@swift-ci please test

**Comment by davezarzycki** [comment]
@swift-ci please clean test linux

**Comment by compnerd** [comment]
Yeah, I agree on the idea that if the user specified it, we assume correct and honor it.

**Comment by davezarzycki** [comment]
@swift-ci please test windows

**Pull request merged**